utils/src/main/scala/za/co/absa/enceladus/utils/types/TypedStructField.scala

```diff
@@ -349,14 +349,21 @@ object TypedStructField {
       DecimalTypeStructField.minPossible(dataType),
       DecimalTypeStructField.maxPossible(dataType)
     ){
+    val strictParsing: Boolean = getMetadataStringAsBoolean(MetadataKeys.StrictParsing).getOrElse(false)
 
     override val parser: Try[DecimalParser] =
       pattern.map { patternOpt =>
         DecimalParser(patternOpt.getOrElse(NumericPattern(defaults.getDecimalSymbols)), Option(typeMin), Option(typeMax))
       }
 
     override def validate(): Seq[ValidationIssue] = {
-      NumericFieldValidator.validate(this)
+      DecimalFieldValidator.validate(this)
+    }
+
+    override protected def convertString(string: String): Try[BigDecimal] = {
+      if (strictParsing && scale < string.split("\\.")(1).length)
+          Failure(new IllegalArgumentException(s"Strict parsing does not allow $string"))
```

Comment:
This message will need an update. It is not strict parsing that doesn't allow random values. It is the `s"Mismatch between the scale of the data and the schema defined scale resulting in loss of precission"` or something like this

Reply:
True, the message should be more explanatory. How about `$string exceeds the defined scale limit in the schema`

Reply:
Even better